packages/components/link/_link-theme.scss

```
@@ -17,7 +17,8 @@
     }
 
     &.kbq-text-only,
```

Предлагаю добавить сюда css-класс и чтобы избавиться от дополнительных вычислений в компоненте, проставлять CSS-класс по месту.
Предлагаю сделать так потому что ссылка может быть не всегда в конце предложения, а `hasDot` будет считаться для всех ссылок

```suggestion
    &.kbq-text-only:not(.kbq-text-with-dot),
```